modules/web-crypto-backend/test/fixtures.ts

```diff
@@ -98,3 +98,145 @@ export const subtleFallbackZeroByteEncryptFail = {
 } as any
 
 export const subtleFallbackNoWebCrypto = {} as any
+
+export const fakeWindowIE11OnComplete = {
+  msCrypto: {
+    getRandomValues: (values: Uint8Array) => {
+      return values.fill(1)
+    },
+    subtle: {
+      decrypt () {
+        const obj = {} as any
+        setTimeout(() => {
+          obj.result = true
+          obj.oncomplete()
+        })
+        return obj
+      },
+      digest () {
+        const obj = {} as any
+        setTimeout(() => {
+          obj.result = true
+          obj.oncomplete()
+        })
+        return obj
+      },
+      encrypt () {
+        const obj = {} as any
+        setTimeout(() => {
+          obj.result = true
+          obj.oncomplete()
+        })
+        return obj
+      },
+      exportKey () {
+        const obj = {} as any
+        setTimeout(() => {
+          obj.result = true
+          obj.oncomplete()
+        })
+        return obj
+      },
+      generateKey () {
+        const obj = {} as any
+        setTimeout(() => {
+          obj.result = true
+          obj.oncomplete()
+        })
+        return obj
+      },
+      importKey () {
+        const obj = {} as any
+        setTimeout(() => {
+          obj.result = true
+          obj.oncomplete()
+        })
+        return obj
+      },
+      sign () {
+        const obj = {} as any
+        setTimeout(() => {
+          obj.result = true
+          obj.oncomplete()
+        })
+        return obj
+      },
+      verify () {
+        const obj = {} as any
+        setTimeout(() => {
+          obj.result = true
+          obj.oncomplete()
+        })
+        return obj
+      }
+    }
+  },
+  MSInputMethodContext: {} as any
+} as any
+
+export const fakeWindowIE11OnError = {
+  msCrypto: {
+    getRandomValues: (values: Uint8Array) => {
+      return values.fill(1)
+    },
+    subtle: {
+      decrypt () {
+        const obj = {} as any
```

Comment:
I don't know much about JavaScript/TypeScript, but I'm noticing that a lot of these function definitions are exactly the same, could you make one function and call it whenever needed?

Reply:
I could.  An this is effectively what I'm doing in the code that uses this.  I often chose to be more verbose in tests so that the test code and the code that is tested are not equally abstract...